### ISSUE TITLE
Using Python 3.10 causes output directory to be deleted.

### ISSUE THREAD

**Issue by onedr0p** [mirkolenz/makejinja#105]
**Describe the bug**

It looks like some people are reporting this issue with using makejinja with Python 3.10. I'm pretty sure you only support 3.11+ so I'm kinda confused how they are even able to install the latest version on 3.10.

I'm not sure if there's something you can do on your end but I wanted to report it here in case others have this issue.

I already have a validation function to check which Python version they have install.

https://github.com/onedr0p/flux-cluster-template/blob/eec9fcf7f0327ee3471185d995965c3ce8cce163/bootstrap/scripts/validation.py#L70L73

**To Reproduce**

Install and run `makejinja --force` on Python 3.10

**Expected behavior**

Output directory to not be removed.

**Additional context**

My config file: https://github.com/onedr0p/flux-cluster-template/blob/main/makejinja.toml

Here's the issue that has some more information

https://github.com/onedr0p/flux-cluster-template/issues/1174

**Comment by mirkolenz** [comment]
Thanks for the heads-up. I read through all comments in the issue you linked to and think the issue is that an older version of makejinja was installed by pip (if no version constraint was given during the install command). The support for Python 3.10 was dropped some time ago and the flag `--force` was added afterwards. Prior to that, the default was to always remove the output directory. At some point I thought that this was a quite dangerous default, so `--force` was added.

TLDR: When using Python 3.10, an old version of makejinja will be installed that removes the output directory by default.

**Comment by mirkolenz** [comment]
Just another thought: This error implies that the makejinja package is installed without version constraints. I am not too familiar with the way the flux cluster template works, but maybe you need to add a constraint like `makejinja==x.y.z` to a `requirements.txt` file? The strange thing is that I remember seeing such a constraint in at least of requirements file in this repo, but I have the feeling that there may be other scripts that try to install makejinja without the right constraint?

**Comment by onedr0p** [comment]
Thanks for the insight and I figured there's nothing to do on your end. Having this issue here should hopefully help someone in the future though.

I do pin the version in requirements but that doesn't stop people from changing it. Maybe I'll add a comment in there saying "Do not downgrade, make sure you are using Python 3.11" or something. 😂

**Comment by mirkolenz** [comment]
Yeah it is sometimes hard to predict all the possible ways something can go wrong 😆
